backend/app/interactem/app/api/routes/pipelines.py

```diff
@@ -61,8 +61,181 @@ def read_pipeline(session: SessionDep, current_user: CurrentUser, id: uuid.UUID)
     if not pipeline:
         raise HTTPException(status_code=404, detail="Pipeline not found")
     if not current_user.is_superuser and (pipeline.owner_id != current_user.id):
-        raise HTTPException(status_code=400, detail="Not enough permissions")
-    return pipeline
+        raise HTTPException(status_code=403, detail="Not enough permissions")
+    return PipelinePublic.model_validate(pipeline)
+
+
+@router.patch("/{id}", response_model=PipelinePublic)
+def update_pipeline(
+    *,
+    session: SessionDep,
+    current_user: CurrentUser,
+    id: uuid.UUID,
+    update: PipelineUpdate,
+) -> Any:
+    """
+    Update a pipeline's name.
+    """
+    pipeline = session.get(Pipeline, id)
+    if not pipeline:
+        raise HTTPException(status_code=404, detail="Pipeline not found")
+    if not current_user.is_superuser and (pipeline.owner_id != current_user.id):
+        raise HTTPException(status_code=403, detail="Not enough permissions")
```

Comment:
I think these two should be combined one case that return `404`. By return `403` for a pipeline that user it not owner of leaks the information that the pipeline exists. Really unless the call is the owner of the pipeline it doesn't exist?